File: LeetCode_Marathons/LCM5/Hard/Consecutive_Numbers_Sum/a.py
class Solution:
    def getAllDivizors(self, n: int) -> List[int]:
        allDivizors: List[int] = []
        sqrtN: int = int(n ** 0.5)
        for i in range(1, sqrtN + 1):
            if n % i == 0:
                allDivizors.append(i)
                if n / i != i: allDivizors.append(int(n / i))
        return allDivizors

    def consecutiveNumbersSum(self, n: int) -> int:
        allDivizors: List[int] = self.getAllDivizors(n)
        counter: int = 0
        for divizor in allDivizors:
            if divizor % 2 == 1: counter += 1
        return counter

File: LeetCode_Marathons/LCM5/Hard/Consecutive_Numbers_Sum/a.go
func getAllDivizors(n int) []int {
    allDivizors := []int{}
    sqrtN := int(math.Sqrt(float64(n)))
    for i := 1; i <= sqrtN; i++ {
        if n % i == 0 {
            allDivizors = append(allDivizors, i)
            if n / i != i {
                allDivizors = append(allDivizors, n / i)
            }
        }
    }
    return allDivizors
}

func consecutiveNumbersSum(n int) int {
    allDivizors := getAllDivizors(n)
    counter := 0
    for _, divizor := range allDivizors {
        if divizor % 2 == 1 {
            counter++
        }
    }
    return counter
}